python/src/scipp/compat/mantid.py

```diff
@@ -52,6 +53,69 @@ def make_run(ws):
     return sc.Variable(value=deepcopy(ws.run()))
 
 
+additional_unit_mapping = {
+    "Kelvin": sc.units.K,
+    "microsecond": sc.units.us,
+    "nanosecond": sc.units.ns,
+    "second": sc.units.s,
+    "Angstrom": sc.units.angstrom,
+    "Hz": sc.units.dimensionless / sc.units.s,
+    "degree": sc.units.deg,
+}
+
+
+def make_variables_from_run_logs(ws):
+    lookup_units = dict(
+        zip([str(unit) for unit in sc.units.supported_units()],
+            sc.units.supported_units()))
+    lookup_units.update(additional_unit_mapping)
+    for property_name in ws.run().keys():
+        units_string = ws.run()[property_name].units
+        units = lookup_units.get(units_string, None)
+        values = deepcopy(ws.run()[property_name].value)
+
+        if units_string and units is None:
+            warnings.warn(f"Workspace run log '{property_name}' "
+                          f"has unrecognised units: '{units_string}'")
+
+        try:
+            times = deepcopy(ws.run()[property_name].times)
+            is_time_series = True
+            dimension_label = "times"
+        except AttributeError:
+            times = None
+            is_time_series = False
+            dimension_label = property_name
+
+        if np.isscalar(values):
+            if units is not None:
+                property_data = sc.Variable(value=values, unit=units)
+            else:
+                property_data = sc.Variable(value=values)
+        else:
+            if units is not None:
+                property_data = sc.Variable(values=values,
+                                            unit=units,
+                                            dims=[dimension_label])
+            else:
+                property_data = sc.Variable(values=values,
+                                            dims=[dimension_label])
```

Comment:
The code here would be significantly shortened by setting `units=sc.units.one` in the failure case above, avoiding two `if units is not None` statements.